test/lint/lint-shell.sh

```diff
@@ -10,6 +10,8 @@
 # to allow running certain versions of shellcheck that core dump when LC_ALL=C
 # is set.
 
+unset LC_ALL
```

Comment:
Can you make this `unset` conditional on the specific environment where this `shellcheck` issue occurs?

Reply:
Why? The comment above even says that it specifically does not set LC_ALL because it seg fauls (core cumps) when it is set. Means that if I have LC_ALL set and exported and run this script it will segfault. So to me it looks as if unsetting LC_ALL for good should be required on all platforms.

I can see how this makes one uneasy (at least it makes me uneasy). According to one of the lints all scripts _should_ set LC_ALL=C and export it in order to make it not behave differently on different platforms. Unfortunately that makes shellcheck segfault/coredump. As I learned the hard way.

I am wondering if there is a specific locale which does not make it segfault, but I would assume that that locale is exactly the default locale for the platform. And as long as none of the shell scripts use any fancy characters (basically ASCII only) it should not come to any behavior changes in between any modern platforms.

Reply:
@scravy I wrote that comment in the code :-)  The comment says that `LC_ALL=C` segfaults under certain `shellcheck` versions (perhaps a certain Ubuntu distributed `shellcheck` binary?), so my suggestions is to make the workaround (`unset LC_ALL`) conditional on where it is actually needed. That way we document when it is needed which will help us make an informed decision on when to remove the workaround in the future, such as when that specific problematic version is no longer in use. Makes sense?

Reply:
I believe on travis just all versions would segfault :/

Reply:
Oh, that's bad! I didn't know that. I thought it was generally for a specific Ubuntu version in combination with a specific `shellcheck` version. No one has been able to reproduce this locally when running same version? @scravy, can you verify that it is not present locally?

Reply:
@practicalswift So I now checked with/and without `export LC_ALL=C` on Ubuntu 14.04, 16.04, 18.04 in docker and in 18.04 on my ubuntu laptop and the shellcheck version that is part of the distro does not segfault (0.3.3, 0.3.7, 0.4.6) respectively.

Looks to me as if in travis they just have a horrible version.

Okay, so I guess I can make this script use `export LC_ALL=C` just like the linter actually wants and unset it in case it is running in travis.

Reply:
@scravy That would be excellent! :-)

Reply:
@practicalswift @MarcoFalke I extracted this change into: https://github.com/bitcoin/bitcoin/pull/13851